sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Storage/BlobsCheckpointStore.cs

```diff
@@ -88,6 +108,11 @@ await foreach (BlobItem blob in ContainerClient.GetBlobsAsync(traits: BlobTraits
                 }
 
                 return ownershipList;
+            };
+
+            try
+            {
+                return ApplyRetryPolicy(listOwnershipAsync, cancellationToken);
```

Comment:
Should we be retrying?  Doesn't the blob client handle retries on our behalf?

Reply:
I guess that it doesn't hurt, given that our goal is to be as resilient as possible and we don't have direct control over the retry configuration of the blob container client.   We will potentially be doing a retry loop here that calls into a function with its own retry loop, thereby doubling the retries. 

Was that something that was discussed when Srikanta was talking over the design?    If we fail out here (retries gone or no retry) what is the behavior?

Reply:
If I recall correctly, that's something that hasn't been discussed with the team.

I think it's safe to assume the blob client applies its own retry attempts when performing operations. However, since the `RetryPolicy` type is not supported by them, I believe we should still do it on our side to honor any policy the user specifies.

Reply:
I'm good with it.   Just wanted to make sure we were making an informed decision.